openmdao/core/system.py

```diff
@@ -1792,7 +1797,13 @@ def update_src_indices(name, key):
                                            (self.msginfo, name, str(flat_src_indices),
                                             str(meta['flat_src_indices'])))
 
-                meta['src_indices'] = np.asarray(src_indices, dtype=INT_DTYPE)
+                if isinstance(src_indices, np.ndarray):
```

Comment:
could add `and src_indices.dtype==object` to the 'if' check since that's the only situation where an ndarray could contain a slice.